web/src/lib/utils/asset-utils.ts

```diff
@@ -438,6 +440,26 @@ export const deleteStack = async (stackIds: string[]) => {
   }
 };
 
+export const keepThisDeleteOthers = async (keepAsset: AssetResponseDto, stack: StackResponseDto) => {
+  const $t = get(t);
+
+  try {
+    const assetsToDeleteIds = stack.assets.filter((asset) => asset.id !== keepAsset.id).map((asset) => asset.id);
+    await deleteStacks({ bulkIdsDto: { ids: [stack.id] } });
+    await deleteAssets({ assetBulkDeleteDto: { ids: assetsToDeleteIds } });
```

Comment:
`deleteAssets` will permanently delete these assets if the trash feature is disabled, so you'll want to display a deletion warning and get user confirmation. There should be other examples of this in the code.

Reply:
Good to know.  I'll look into doing this!

Reply:
Confirmation dialog has been implemented when clicking "Keep this, delete others."
![image](https://github.com/user-attachments/assets/a917f880-2077-4118-ad8f-04e51125482a)